File: java to c++/Broadcast.cpp
#include "Broadcast.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						using Response = bgu::spl171::net::impl::TFTP::Commands::Responses::Response;

						Broadcast::Broadcast(bool isAdded, const std::wstring &fileName)
						{
							this->isAdded_Renamed = isAdded;
							this->fileName = fileName;
						}

						bool Broadcast::isAdded()
						{
							return isAdded_Renamed;
						}

						std::wstring Broadcast::getFileName()
						{
							return fileName;
						}

						short Broadcast::getOpCode()
						{
							return OpCode::Broadcast::getValue();
						}
					}
				}
			}
		}
	}
}

File: Server/src/main/java/bgu/spl171/net/impl/TFTP/Commands/Responses/Broadcast.java
package bgu.spl171.net.impl.TFTP.Commands.Responses;

import bgu.spl171.net.impl.TFTP.Commands.Command;
import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class Broadcast implements Response {
    boolean isAdded;
    String fileName;

    public Broadcast(boolean isAdded, String fileName) {
        this.isAdded = isAdded;
        this.fileName = fileName;
    }

    public boolean isAdded() {
        return isAdded;
    }

    public String getFileName() {
        return fileName;
    }

    @Override
    public short getOpCode() {
        return OpCode.Broadcast.getValue();
    }

}
